stac_fastapi/types/stac_fastapi/types/core.py

```diff
@@ -80,6 +81,100 @@ def update_item(
         """
         ...
 
+    def patch_item(
```

Comment:
What do you think about making `patch_item` the abstract method. Having `patch_item` depending on two abstract method makes things a bit complex IMO and will make customization a bit difficult.

Reply:
Yes that sounds sensible to me. I'll make the changes now!